packages/quantic/force-app/main/default/lwc/quanticRecommendationList/__tests__/quanticRecommendationList.test.js

```diff
@@ -0,0 +1,428 @@
+            },
+          }
+        );
+      });
+    });
+  });
+
+  describe('when an initialization error occurs', () => {
+    beforeEach(() => {
+      mockErroneousHeadlessInitialization();
+    });
+
+    it('should display the error component', async () => {
+      const element = createTestComponent();
+      await flushPromises();
+
+      const initializationError = element.shadowRoot.querySelector(
+        selectors.initializationError
+      );
+
+      expect(initializationError).not.toBeNull();
+    });
+  });
+
+  describe('when the component is initialized successfully', () => {
+    beforeEach(() => {
+      mockSuccessfulHeadlessInitialization();
+      prepareHeadlessState();
+    });
+
+    describe('when there are no recommendations', () => {
+      it('should not display a recommendation list', async () => {
+        recommendationListState = {
+          isLoading: false,
+          recommendations: [],
+          error: null,
+          searchResponseId: null,
+        };
+        const element = createTestComponent();
+        await flushPromises();
+
+        const recommendationContainer = element.shadowRoot.querySelector(
+          selectors.gridContainer
+        );
+
+        expect(recommendationContainer).toBeNull();
+      });
+    });
+
+    describe('when there are recommendations', () => {
+      describe.each([
+        ['grid', selectors.gridContainer, selectors.recommendationListItem],
+        [
+          'carousel',
+          selectors.carouselContainer,
+          selectors.recommendationListItem,
+        ],
+      ])('with %s variant', (variant, containerSelector, itemSelector) => {
+        beforeEach(() => {
+          recommendationListState = {
+            isLoading: false,
+            recommendations: recommendationsArray,
+            error: null,
+            searchResponseId: '123',
+          };
+        });
+        it('should properly display the recommendations with default options', async () => {
+          const element = createTestComponent({
+            ...defaultOptions,
+            variant,
+          });
+          await flushPromises();
+
+          const recommendationContainer =
+            element.shadowRoot.querySelector(containerSelector);
+          const recommendationItems =
+            element.shadowRoot.querySelectorAll(itemSelector);
+
+          expect(recommendationContainer).not.toBeNull();
+          if (variant === 'grid') {
+            expect(recommendationItems.length).toBe(
+              recommendationsArray.length
+            );
+          } else {
+            expect(recommendationContainer.label).toEqual(defaultOptions.label);
+            expect(recommendationContainer.itemsPerPage).toEqual(
+              defaultOptions.recommendationsPerRow
+            );
+          }
+        });
+
+        describe('with a custom #recommendationsPerRow option', () => {
+          const customRecommendationsPerRow = 2;
+          if (variant === 'grid') {
```

Comment:
Not sure I'm a fan of the `describe.each` if you have to end up doing this `if`.

I think I would split with a `describe('#variant : grid')` and `describe('#variant : caroussel')` separate.

Reply:
yeah I tried to get fancy with the each but I was on the fence on this one. Will split the tests!